faq.md

```diff
@@ -112,162 +111,136 @@ It’s a component that was introduced recently. Essentially it’s a stateless
-There’s not currently an option for “infinite” (though it sounds a good idea! maybe we could use `-1` for that). The only option now is to use INT_MAX for `retentionTimeInMinutes` and LONG_MAX for `retentionSizeInMB`. It’s not “infinite” but 4085 years of retention should probably be enough!
+A third option is to check the topic stats. This is a tiny bit involved, because it requires the admin client (or using REST) to get the stats for the topic and checking the "backlog". If the backlog is 0, it means you've hit the end.
 
-### Is there a profiling option in Pulsar, so that we can breakdown the time costed in every stage? For instance, message A stay in queue 1ms, bk writing time 2ms(interval between sending to bk and receiving ack from bk) and so on.
-There are latency stats at different stages. In the client (eg: reported every 1min in info logs). 
-In the broker: accessible through the broker metrics, and finally in bookies where there are several different latency metrics. 
+### How can I prevent an inactive topic to be deleted under any circumstance? I do not want to set time or space limit for a certain namespace.
+You can set *infinite* retention time or size, by setting `-1` for either time or
+size retention.
+For more details, see [Pulsar retention policy](http://pulsar.apache.org/docs/en/cookbooks-retention-expiry/#retention-policies).
 
-In broker there’s just the write latency on BK, because there is no other queuing involved in the write path.
+### Is there a profiling option in Pulsar, so that we can breakdown the time costed in every stage? For instance, message A stays in queue 1ms, bookkeeper writes in 2ms(interval between sending to bk and receiving ack from bk) and so on.
+There are latency stats at different stages: client, bookie and broker. 
+For example, the client reports every minute in info logs. 
+The broker is accessible through the broker metrics, and bookies store several different latency metrics. 
 
-### How can I have multiple readers that each get all the messages from a topic from the beginning concurrently? I.e., no round-robin, no exclusivity
-you can create reader with `MessageId.earliest`
+In broker, there’s just the write latency on Bookkeeper, because there is no other queuing involved in the write path.
 
+### How can multiple readers get all the messages from a topic from the beginning concurrently? I.e., no round-robin, no exclusivity.
+You can create reader with `MessageId.earliest`.
 
 ### Does broker validate if a property exists or not when producer/consumer connects ?
-yes, broker performs auth&auth while creating producer/consumer and this information presents under namespace policies.. so, if auth is enabled then broker does validation
+Yes, broker performs auth&auth while creating producer/consumer and this information presents under namespace policies. So, if auth is enabled, and then broker performs validation.
```

Comment:
"Property" should be "tenant" now. 

Answer is unclear. auth & auth is authentication and authorization, but which one needs to be enabled for validation.

Reply:
@sijie @jiazhai could you please confirm which auth needs to be enabled?